### PR TITLE
fix: Adds pagination to Facility linked list API

### PR DESCRIPTION
**Pull request by aeswibon**
# Bug Fix

## Proposed Changes

- [x] Adds `pagination` in linked facilities list API

## Associated Issue

- Required by coronasafe/care_fe#7257

## Merge Checklist

- [x] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/users/api/viewsets/users.py [hunk 1]
```diff
@@ -70,6 +71,10 @@ def get_user_type(
     user_type = filters.CharFilter(method="get_user_type", field_name="user_type")
 
 
+class UserFacilitiesPagination(LimitOffsetPagination):
+    page_size = 20
+
+
```
Comment (sainak) [inline care/users/api/viewsets/users.py:77]:
```suggestion
```
let's use the default pagination class. We can always specify the page size the fe wants

File: care/users/api/viewsets/users.py [hunk 2]
```diff
@@ -100,6 +105,7 @@ class UserViewSet(
         rest_framework_filters.OrderingFilter,
         drf_filters.SearchFilter,
     )
+    pagination_class = UserFacilitiesPagination
```
Comment (sainak) [inline care/users/api/viewsets/users.py:108]:
```suggestion
```

**Review by sainak — CHANGES_REQUESTED** [review]

File: care/users/api/viewsets/users.py (hunk 2 above)
Comment (aeswibon) [reply care/users/api/viewsets/users.py:108]: OKay

File: care/users/api/viewsets/users.py (hunk 1 above)
Comment (aeswibon) [reply care/users/api/viewsets/users.py]: Done

**Review by sainak — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**